zh/before-start_en.md

```diff
@@ -0,0 +1,62 @@
+# Before the beginning
+## The target audience for this document
+### The confused software developers
+* Developers who want to write code that can be easily read but don't know how to do that  
+* Developers who want to understand other people's code  
+### Software development beginner
+This is a document for software development beginners, who can be defined as having learned no more than two languages and being able to complete most of exercises in the course in terms of the language skills they have learned. From the general situation of college education, they could basicly complete the final course assignments arranged by teachers. We may not be able to say that they have done very well. It is also difficult to improve further.  
+
+For students who have graduated from college(plus or minus one or two years), it is difficult for them to get access to truly complex projects, and they will be incharge of the detailed parts even if they are involved in complex projects. They are eager to improve their software development ability quickly, but by accident, what they lack most is the opportunity to improve their ability.

```

Comment:
For students who graduated or will graduate from universities(within two years), it is difficult to touch real complex projects, and even if they are involved in complex projects, they will do the trivial details. They are eager to improve their developing ability as soon as possible, but what they lack mostly just is the opportunity to improve themselves.